### ISSUE TITLE
Contributing to the mazes repository

### ISSUE THREAD

**Issue by Peque** [micromouseonline/micromouse_maze_tool#3]
As you have seen, I have just tried to contribute my first maze to this repository. In order to do so I needed to install a C++ compiler, compile the project and generate the 3 different formats required.

I also thought about cleaning up some maze file names (i.e.: make some file names more consistent), but then realized I had to do so for the 3 formats and lost interest... :joy:

Maybe this could be approached in a different way: have a separate repository just for maze files. In my opinion this repository should:

- Contain only maze files, no scripts, no source code.
- Maze files could be stored in folders by maze dimension. For now there would only be 16x16 mazes, so just one folder.
- Contain a single file for each maze (i.e.: avoid duplication). I think this would be the most important part, actually.
- The file format should be text because:
  - It is self-explanatory (i.e.: no need to explain how information is stored in bytes).
  - It is language-agnostic.
  - It gets along with version control systems (Git).
  - It is easy to visualize right from the browser or locally with any text editor.
  - It is also easy to edit (in case someone wants to contribute new mazes or fix existing ones, they only need a text editor if they do not have better/automated tools).
  - It is not hard to parse nor to generate in any programming language.

Then you could use that repository as a Git sub-module in this maze tool or anywhere else you use maze files. I would do the same of course if that repository existed. :blush:

Do you think it could be a good idea?

**Comment by micromouseonline** [comment]
Thanks for the update.

Git submodules always give me trouble. But that says more about me than git.

The maze file names are poor and I keep meaning to sort them out myself but, as you say - - - 

Actually, it is only necessary to change the binary names and then use the software to re-generate the c and text files.

Your ideas about the maze files are all valid.

I regard the binary files as the canonical versions and use them to create the others. It would be equally valid to use the text form and there are many advantages. Perhaps that is the way to go.

Peter Harrison
peter.harrison@helicron.net

**Comment by micromouseonline** [comment]
I have created a repository containing only micromouse maze files in text format.

Then names have been rationalised a bit and a number of mazes have been removed where they are clearly erroneous or likely incorrect duplicates of other mazes.

https://github.com/micromouseonline/mazefiles

Now I will think about updating the mazetool repo to point to these

**Comment by Peque** [comment]
That is great news, thanks! :smile:
